src/components/CardListSection.jsx

```diff
@@ -0,0 +1,19 @@
+import NoPage from "./NoPage";
+import useSelectFolder from "../hooks/useSelectFolder";
+import CardList from "./CardList";
+
+function CardListSection({ url, folders }) {
+  const result = useSelectFolder({ url });
+  const links = result?.data;
+  return (
+    <>
+      {links && links.length === 0 ? (
+        <NoPage />
+      ) : (
+        <CardList folders={folders} links={links} />
+      )}
+    </>
+  );
```

Comment:
```suggestion
      links && links.length === 0 ? (
        return <NoPage />
      ) : (
	return <CardList folders={folders} links={links} />
      )
  );
```

이렇게도 가능합니다.

컴포넌트는 JSX만 return하면 되는 함수라고 생각해주시면 편합니다.